pkgs/build-support/dart/build-dart-application/hooks/dart-install-hook.sh

```diff
@@ -0,0 +1,31 @@
+# shellcheck shell=bash
+
+dartInstallHook() {
+    echo "Executing dartInstallHook"
+
+    runHook preInstall
+
+    mkdir -p "$out"
+
+    while IFS=$'\t' read -ra target; do
+        dest="${target[0]}"
+        # Wrap with runtime command, if it's defined
+        if [ -n "$dartRuntimeCommand" ]; then
+            install -D "$dest" "$out/share/$dest"
+            install -D /dev/null "$out/$dest"
+cat > "$out/$dest" <<EOF
```

Comment:
Oh that's a good point! I was also thinking of doing that for https://github.com/NixOS/nixpkgs/pull/228931/files#diff-7f45bff25a681f238dff3bfe6b2cc2d15a4383e5912d8d58475dbe0f9c7e7c33R61-R62, and ended up using echo because I thought `makeWrapper` didn't have a "prefix binary" option.

But just thinking about it again, I was going about it the wrong way. You could just use `makeWrapper` on the runtime command binary, and just pass `$dest` as an argument.

Reply:
Yeah that's the problem I had too. I remember I considered using `add-flags` like Kira mentioned, but for some reason didn't. I'll fix that in a sec

Reply:
Oh I remember now! It's because I need to wrap `jit` and `kernel` with `dart run` specifically. And makeWrapper does not work with that (that's why `$dartRuntimeCommand` is unquoted).

To use `add-flags`, I'd had to conditionally add them based on `dartRuntimeCommand` having spaces or not, and I think that's kind of a pain. What do you folks think?

Reply:
Oh wait a second though, a shebang using `runtimeShell` should be added though.

Reply:
Actually, it seems using just `dart` to run them also works. I'll use `wrapProgram` instead.

Reply:
And `\$@` should also be quoted.